### PR TITLE
Add hotplug/unplug test for rng_device for backend rdm,tcp,udp

### PR DESCRIPTION
**Pull request by kylazhang**
Signed-off-by: Kylazhang <weizhan@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/libvirt_rng.py [hunk 1]
```diff
@@ -61,6 +107,8 @@ def modify_rng_xml(dparams, sync=True):
         rng_xml.backend = backend
 
         logging.debug("Rng xml: %s", rng_xml)
+        if get_xml:
+            return rng_xml
```
Comment (chloerh) [inline libvirt/tests/src/libvirt_rng.py]: If 'get_xml' is True, 'sync' could still be True. Maybe line:110-111 should be moved after the 'sync' part

File: libvirt/tests/src/libvirt_rng.py [hunk 2]
```diff
@@ -29,9 +30,54 @@ def run(test, params, env):
...
+            xml_get = xml_devices[rng_index]
+
+            if not exists:
+                test.fail("Rng device still exists after detach!")
+        except IndexError:
+            if exists:
+                test.fail("Can not find rng device in xml")
+            else:
+                logging.info("Can not find rng xml as expected")
+                return
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_rng.py]: no need

File: libvirt/tests/src/libvirt_rng.py [hunk 3]
```diff
@@ -29,9 +30,54 @@ def run(test, params, env):
...
+
+        match = 1
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_rng.py]: match = True?

File: libvirt/tests/src/libvirt_rng.py (hunk 1 above)
Comment (kylazhang) [reply libvirt/tests/src/libvirt_rng.py]: Here the logic is when get_xml is True, will not sync to guest. I have already update the function docstring.

File: libvirt/tests/src/libvirt_rng.py (hunk 3 above)
Comment (kylazhang) [reply libvirt/tests/src/libvirt_rng.py]: fixed, thanks

**Pull request merged**